apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerHealthChecker.java

```diff
@@ -85,7 +85,7 @@ public Version withConnection(HttpURLConnection connection) {
                     } else {
                         try {
                             // prints out the version info of the APM Server
-                            String body = HttpUtils.getBody(connection);
+                            String body = HttpUtils.readToString(connection.getInputStream());
```

Comment:
@felixbarny we may be missing valuable info from the error stream, but we can log that when consuming and closing (see below). WDYT?

Reply:
> we may be missing valuable info from the error stream

You mean you would like to log the APM Server response?

If there is an error stream, `connection.getResponseCode()` would have thrown an exception. But we could check in the outer catch if there's an error and log it. I just didn't want to change the logic in this PR

Reply:
I raised this because this was the logic before this PR (through `getBody`) 🙂

Reply:
Got it. I removed it because that code path could never be reached and even if, it would not make much sense.

If there was an error stream, we would not even be in this branch, as `connection.getResponseCode()` (called on line 77) would have thrown an exception. There's also another if branch for status codes >= 300 so in this branch, there are only successful responses.

But even if there was an error stream, it would not really make sense to log in the next line `logger.info("Elastic APM Server is available: {}", body);` as an error stream would indicate a non-available APM Server.